src/ui/projects/CreateScenePage.js

```diff
@@ -0,0 +1,366 @@
+import { withApi } from "../contexts/ApiContext";
+import NavBar from "../navigation/NavBar";
+import Footer from "../navigation/Footer";
+import styled from "styled-components";
+
+import React, { useState, useCallback, useEffect } from "react";
+import { useHistory } from "react-router-dom";
+
+import StringInput from "../inputs/StringInput";
+import BooleanInput from "../inputs/BooleanInput";
+import FileInput from "../inputs/FileInput";
+import FormField from "../inputs/FormField";
+import { Button } from "../inputs/Button";
+import ProgressBar from "../inputs/ProgressBar";
+
+const SceneUploadFormContainer = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: row;
+  background-color: ${props => props.theme.panel2};
+  border-radius: 3px;
+`;
+
+const ErrorMessage = styled.div`
+  color: ${props => props.theme.red};
+  margin-top: 8px;
+`;
+
+export const UploadSceneSection = styled.form`
+  padding-bottom: 100px;
+  display: flex;
+
+  &:first-child {
+    padding-top: 100px;
+  }
+
+  h1 {
+    font-size: 36px;
+  }
+
+  h2 {
+    font-size: 16px;
+  }
+`;
+
+export const UploadSceneContainer = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  margin: 0 auto;
+  max-width: 800px;
+  min-width: 400px;
+  padding: 0 20px;
+`;
+
+export const SceneUploadHeader = styled.div`
+  margin-bottom: 36px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const LeftContent = styled.div`
+  display: flex;
+  width: 360px;
+  border-top-left-radius: inherit;
+  align-items: flex-start;
+  padding: 30px;
+  position: relative;
+
+  img,
+  div {
+    width: 300px;
+    height: 168px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    text-align: center;
+    background-color: ${props => props.theme.panel};
+    border-radius: 6px;
+  }
+  input {
+    opacity: 0;
+    position: absolute;
+  }
+`;
+
+const RightContent = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  padding: 30px 30px;
+
+  label[type="button"] {
+    display: flex;
+    margin-bottom: 0;
+    margin-right: 5px;
+  }
+`;
+
+function CreateScenePage({ match, api }) {
+  const [isUploading, setIsUploading] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [sceneId, setSceneId] = useState(null);
+  const [error, setError] = useState(null);
+
+  const [sceneInfo, setSceneInfo] = useState({
+    name: "",
+    creatorAttribution: "",
+    allowRemixing: false,
+    allowPromotion: false
+  });
+
+  const [glbFile, setGlbFile] = useState(null);
+  const [thumbnailFile, setThumbnailFile] = useState(null);
+
+  const [thumbnailUrl, setThumbnailUrl] = useState(null);
+  const [sceneUrl, setSceneUrl] = useState(null);
+
+  const history = useHistory();
+
+  const openScene = useCallback(
+    function() {
+      window.open(sceneUrl);
+    },
+    [sceneUrl]
+  );
+
+  useEffect(
+    function() {
+      async function doInitialLoad() {
```

Comment:
This doesn't need to be an arrow function since it's a named function. It does need a try/catch block or some `catch` handler on the async function invocation though.